### ISSUE TITLE
BUG: gradient Explainer pd.DataFrame input failing

### ISSUE THREAD

**Issue by ponseko** [shap/shap#3152]
### Issue Description

The gradient explainer class accepts `pandas.Dataframe` as a data input for TF models. The dataframe is correctly used for its feature_name columns, but is then passed on as a `pd.Dataframe` to `_TFGradient`, which does not deal with this type properly.
A simple fix would be to convert the Dataframe to its underlying np.array

### Minimal Reproducible Example

```python
import shap
    import pandas as pd
    import tensorflow as tf

    feature_names = [f"feature_{i}" for i in range(5)]

    background = pd.DataFrame(np.zeros((10, 5)), columns=feature_names)

    model = tf.keras.Sequential([
        tf.keras.layers.Dense(10, input_shape=(5,), activation='relu'),
        tf.keras.layers.Dense(1, activation='linear')
    ])
    model.compile(optimizer='adam', loss='mse')
    e = shap.GradientExplainer(model, background)
```

### Expected Behavior

The `GradientExplainer` should deal with pd.Dataframes as input as is described.

### Bug report checklist

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest release](https://github.com/shap/shap/releases) of shap.
- [X] I have confirmed this bug exists on the [master branch](https://github.com/shap/shap/blob/master/CONTRIBUTING.md#installing-from-the-master-branch) of shap.
- [X] I'd be interested in making a PR to fix this bug

### Installed Versions

0.42.1
